### PR TITLE
button: support tooltip

### PR DESCRIPTION
**Pull request by muhajirdev**
### Summary
Add tooltip support to button

### Screenshots

Before

image

After

image

### CODE REVIEW

**Comment by mmahalwy** [comment]
#### 📝 Changed routes:


| [**/search**](https://quran-com-frontend-next-ldo7t7x4r-qurancom.vercel.app/search) |
|:-:|
|<a href="https://quran-com-frontend-next-ldo7t7x4r-qurancom.vercel.app/search">Screenshot of /search</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/ZSbSlB4IE6obiNm4FqXa5A2kEHjeezSdQnOXThRSon8PpOzoRYqVJIsp8mqhgUJhm-oWW9KK2S4cWkj2tSszRQ.png">(view full size)</a> |







Commit 4df5acf9b1422109f90420c0f0b5825218bc2590 (https://quran-com-frontend-next-ldo7t7x4r-qurancom.vercel.app).

File: src/components/AudioPlayer/AudioPlayer.tsx [hunk 1]
```diff
@@ -140,6 +140,7 @@ const AudioPlayer = () => {
           <PlayPauseButton />
           <div className={styles.seekBackwardsContainer}>
             <Button
+              tooltip="seek -10s"
```
Comment (osamasayed) [inline src/components/AudioPlayer/AudioPlayer.tsx:143]: I feel `seek` is too technical. Should it be "Fast-forward 10 seconds" instead?

File: src/components/AudioPlayer/PlaybackControls.tsx [hunk 2]
```diff
@@ -17,10 +17,17 @@ const PlaybackControls = () => {
...
+        tooltip="feature coming soon"
+        variant={ButtonVariant.Ghost}
+        shape={ButtonShape.Circle}
+        size={ButtonSize.Large}
+        disabled
+      >
         <RepeatIcon />
       </Button>
       <Button
+        tooltip="seek -10s"
```
Comment (osamasayed) [inline src/components/AudioPlayer/PlaybackControls.tsx:30]: This should be `Rewind 10 seconds`

File: src/components/AudioPlayer/PlaybackControls.tsx [hunk 3]
```diff
@@ -31,6 +38,7 @@ const PlaybackControls = () => {
       </Button>
       <PlayPauseButton />
       <Button
+        tooltip="seek +10s"
```
Comment (osamasayed) [inline src/components/AudioPlayer/PlaybackControls.tsx:41]: See my comment above for `seek +10s`

File: src/components/AudioPlayer/AudioPlayer.tsx [hunk 4]
```diff
@@ -161,12 +162,12 @@ const AudioPlayer = () => {
         {/* The div below serves as placeholder for a right section, as well as for centering the slider */}
         <div className={styles.rightSection}>
           {visibility === Visibility.Expanded && (
-            <Button shape={ButtonShape.Circle} variant={ButtonVariant.Ghost}>
+            <Button tooltip="minimize" shape={ButtonShape.Circle} variant={ButtonVariant.Ghost}>
```
Comment (osamasayed) [inline src/components/AudioPlayer/AudioPlayer.tsx:165]: I think all tooltips should have the values capitalized e.g. `minimize`-> `Minimize`

File: src/components/dls/Button/Button.tsx [hunk 5]
```diff
@@ -97,11 +101,16 @@ const Button: React.FC<ButtonProps> = ({
     );
 
   return (
-    <button type="button" className={classes} disabled={disabled} onClick={onClick}>
-      {prefixFinal && <span className={styles.prefix}>{prefixFinal}</span>}
-      <span className={styles.content}>{children}</span>
-      {suffix && <span className={styles.suffix}>{suffix}</span>}
-    </button>
+    <Wrapper
+      condition={!!tooltip}
```
Comment (osamasayed) [inline src/components/dls/Button/Button.tsx:105]: should be changed to `shouldWrap` or `applyWrapping`

File: src/components/dls/Wrapper/Wrapper.tsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+ * @param props
+ * @param props.children - The children to wrap.
+ * @param props.condition - The condition to wrap the children with the wrapper.
+ * @param props.wrapper - The wrapper component.
+ *
+ * Reference:
+ * - https://blog.hackages.io/conditionally-wrap-an-element-in-react-a8b9a47fab2ng.html
+ * - https://arjayosma.com/how-to-conditionally-wrap-a-react-component/
+ */
+const Wrapper = ({ children, condition, wrapper }) => (condition ? wrapper(children) : children);
```
Comment (osamasayed) [inline src/components/dls/Wrapper/Wrapper.tsx:12]: `condition` should be change to either `shouldWrap` or `applyWrapping`

**Pull request merged**